### PR TITLE
make genPackages

### PR DESCRIPTION
**Pull request by Artturin**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -18,48 +18,61 @@
...
-          waylandPkgs = rec {
+          githubTemplate = { attrName, extra ? { } }: prev.callPackage ./templates/github-template.nix { inherit prev attrName extra; };
+          checkMutuallyExclusive = lib.mutuallyExclusive (map (e: e.attrName) attrsExtraChangesNeeded) (map (e: e.attrName) attrsNoExtraChangesNeeded);
+          genPackagesGH =
+            if checkMutuallyExclusive then
+              lib.listToAttrs
+                (map
+                  (a: {
+                    name = a.attrName;
+                    value = githubTemplate a;
```
Comment (Artturin) [inline flake.nix:29]: TODO: think of a way to have gitlab and sourcehut templates.

maybe just have the nix equivalent of a match block in template.nix which uses the domain field in metadata.nix

**Comment by colemickens** [comment]
The update script relies on `repo_git` and `repo_hg` attributes to know whether to try to update using git or mercurial.

glpaper's is wrong, for example. It might be worth removing `repo_git` and `repo_hg` and using the `type` field you've added.

**Comment by Artturin** [comment]
> The update script relies on `repo_git` and `repo_hg` attributes to know whether to try to update using git or mercurial.
> 
> glpaper's is wrong, for example. It might be worth removing `repo_git` and `repo_hg` and using the `type` field you've added.

fixed, let's revisit using the type field in a future pr

**Comment by colemickens** [comment]
We'll still need the composed URL to hand to `git ls-remote` etc, so it certainly feels like a ideological/cleanup fix, but I won't stop you if you want to follow up with it. Otherwise this looks good. Everything builds when I peel back the {current messy set}.

Will merge in a minute, one last look.

**Comment by colemickens** [comment]
I guess I might consider hoisting out some of the extra helpers and functions since they don't need `prev` or `final`, but that feels petty/stylistic, I could see someone arguing its only needed in the context of waylandOverlay anyway...

**Comment by colemickens** [comment]
This is great, thanks @Artturin, lez go.

**Pull request merged**